src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -757,6 +757,25 @@ removeExpiredNFTokenOffers(
     }
 }
 
+static void
+removeDeletedTrustLines(
+    ApplyView& view,
+    std::vector<uint256> const& trustLines,
+    beast::Journal viewJ)
+{
+    std::size_t removed = 0;
+
+    for (auto const& index : trustLines)
+    {
+        if (auto const sleState = view.peek({ltRIPPLE_STATE, index}))
+        {
+            deleteAMMTrustLine(view, sleState, std::nullopt, viewJ);
+            if (++removed == maxDeletableAMMTrustLines)
```

Comment:
No reason to use a counter here we know how many entries are in the vector when the function starts. If there are more than maxDeletableAMMTrustLines entries this is an error because the transactor should never have tried to delete more than this?
You should ensure that you tune maxDeletableAMMTrustLines such that metadata will never be oversized.

Reply:
It should not delete more. What if it does happen? Shouldn't we try to delete as many as we deem safe?

Reply:
No it's a logic error. This code should not proceed in an unsafe state.